File: Java/src/main/java/QuestionBank/Solution701_800/Solution746.java
package QuestionBank.Solution701_800;

/**
 * @author windows
 */
public class Solution746 {
    public int minCostClimbingStairs(int[] cost) {
        int[] df = new int[cost.length + 1];
        df[0] = 0;
        df[1] = 0;
        for (int i = 2; i < df.length; i++) {
            df[i] = Math.min(df[i - 1] + cost[i - 1], df[i - 2] + cost[i - 2]);
        }
        return df[cost.length];
    }


    public static void main(String[] args) {
        Solution746 solution746 = new Solution746();
    }
}

File: C++/src/Solution701_800/Solution746.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution746.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

int Solution::minCostClimbingStairs(vector<int> &cost) {
    int n = cost.size();
    vector<int> dp(n + 1);
    dp[0] = dp[1] = 0;
    for (int i = 2; i < n + 1; ++i) {
        dp[i] = min(dp[i - 2] + cost[i - 2], dp[i - 1] + cost[i - 1]);
    }
    return dp[n];
}
